### ISSUE TITLE
Some questions about SoftmaxCrossEntropyLoss

### ISSUE THREAD

**Issue by yingtongxiong** [Dao-AILab/flash-attention#539]
Hello, I have some questions about this function [https://github.com/Dao-AILab/flash-attention/blob/main/flash_attn/losses/cross_entropy.py#L18C1-L18C7](https://github.com/Dao-AILab/flash-attention/blob/main/flash_attn/losses/cross_entropy.py#L18C1-L18C7). And I consider that this function could do softmax operation for logits according to the func name. I don't know my understanding is correct or not. However, I found that the sum of logits after computing loss is not one and I am doubted that there is no softmax operation conducted in logits. So, is there any misunderstood?

**Comment by tridao** [comment]
It's supposed to do the same as pytorch's CrossEntropyLoss. It doesn't compute softmax (e.g. it returns a loss, not probabilities).

**Comment by yingtongxiong** [comment]
Okay, thank you for your reply. I am very curious that why this function is called SoftmaxCrossEntropyLossFn. And I have misunderstood that this function could do softmax for logits.  And I have noticed the there is a parameter called inplace_backward, so what does this parameter can do?

**Comment by tridao** [comment]
For historical reason, the apex [version](https://github.com/NVIDIA/apex/blob/master/apex/contrib/xentropy/softmax_xentropy.py) that we built on calls it SoftmaxCrossEntropyLoss.

`inplace_backward` will save some memory.

**Comment by yingtongxiong** [comment]
So, if I set the inplace_backward is True. we notice that the logits is connected to the loss. If loss have some changes in backward, then logits will change too. Thank you very much!
